### DISCUSSION TITLE
How to get workbooks or Finops-tool kit to work with MSP

### DISCUSSION THREAD

**Discussion by Jude-Clermont** [microsoft/finops-toolkit#990] [Q&A] [answered]
I work for several Managed Service Providers (MSPs), and part of my role involves generating FinOps reports for multiple clients. We access each client’s tenant through the partner-portal to view Cost Management and Billing details.

However, when I try to use the Cost Optimization Workbook, I am unable to select the partner-portal as a source to view cost data for each client's tenant.

Could you please guide me on how I can use the Cost Optimization Workbook for clients accessed via the partner portal? Any help on achieving this or workarounds would be greatly appreciated.

Thank you!

**Comment by arthurclares** [comment, marked as answer]
@Jude-Clermont Unfortunately, the workbook does not work cross tenant and/or through the Partner portal. The only solution here is for you to use the FinOps Hubs and leverage a feature called "Remote Hubs" which allows you to connect to multiple tenants.
We have in our roadmap to develop a Recommendations dashboard that could be use as a replacement for the workbook in this case. Unfortunately, I don't have any deadline for the Recommendations dashboard to be published at the moment.
